python3Packages.click: 8.1.8 -> 8.2.1

dotlambda opened the pull request:
The [previous PR](https://github.com/NixOS/nixpkgs/pull/425929) was reverted because it breaks click-repl.
This PR circumvents that problem by removing from `python3Packages.celery` the functionality that depends on click-repl.
In order still give access to the affected AMQP REPL, we introduce a top-level attribute `celery` which uses an overridden Python package set. This however violates https://github.com/NixOS/nixpkgs/pull/421201.
I also don't like the number of packages that have to be rebuilt just to build `celery`. Maybe we shouldn't add the top-level attribute and let our users override click if they need the AMQP REPL?



fixes https://github.com/NixOS/nixpkgs/issues/438287
cc @MultisampledNight @caniko @RaitoBezarius @frogamic 

This breaks at least:

- [ ] python3Packages.greynoise @mbalatsko 
- [x] python3Packages.typer @winpat 
- [x] todoman https://github.com/pimutils/todoman/issues/596 @leenaars @antonmosich 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda committed:
python3Packages.click: 8.1.8 -> 8.2.1

Diff: https://github.com/pallets/click/compare/8.1.8...8.2.1

Changelog: https://github.com/pallets/click/blob/8.2.1/CHANGES.rst

dotlambda committed:
python3Packages.click-repl: mark broken

dotlambda committed:
python3Packages.celery: don't include AMQP REPL by default

The REPL depends on click-repl, which doesn't support click >= 8.2.0.

dotlambda committed:
todoman: make click-repl an optional dependency

dotlambda committed:
python3Packages.pytest-celery: fix typo

dotlambda committed:
python3Packages.pytest-celery: don't depend on celery and setuptools

dotlambda committed:
python3Packages.celery: add optional-dependencies.auth

caniko commented:
I know it is out of scope to mention it here, but I have been curious if `pyproject-nix` [definitions](https://pyproject-nix.github.io/pyproject.nix/use-cases/pyproject.html) would be able to solve the issues related to override if we integrate it into nixpkgs. Every python project that makes it into nixpkgs should be defined this way IMO.

dotlambda committed:
python3Packages.greynoise: don't include REPL by default

The REPL depends on click-repl, which doesn't support click >= 8.2.0.

antonmosich commented:
`todoman` also encounters the error from `test_sane_config` when starting the resulting executable with no arguments/subcommand (with my config at least). I don't have the time to fix this right now though, sorry.

dotlambda committed:
python3Packages.flasgger: fix tests with Click 8.2

dotlambda committed:
python3Packages.typer: 0.17.4 -> 0.19.2

Diff: https://github.com/fastapi/typer/compare/0.17.4...0.19.2

Changelog: https://github.com/tiangolo/typer/releases/tag/0.19.2
